spce/spce.go

```diff
@@ -0,0 +1,118 @@
+package spce
+
+import (
+	"bufio"
+	"fmt"
+	"io"
+	"strconv"
+	"strings"
+)
+
+type Code uint8
+
+const (
+	Current  Code = 0x0a
+	Pressure Code = 0x0b
+	Voltage  Code = 0x0c
+)
+
+type Status string
+
+const (
+	Ok Status = "OK"
+	Er Status = "ER"
+)
+
+type SPCE struct {
+	rw      io.ReadWriter
+	channel uint8
+}
+
+// NewSPCE returns a new SPCE instance on a given ReadWriter for the specified channel.
+func NewSPCE(rw io.ReadWriter, channel uint8) *SPCE {
+	return &SPCE{rw: rw, channel: channel}
+}
+
+// Read writes a read command for code c to the SPCE and returns the response data.
+func (s *SPCE) Read(c Code) (string, error) {
+	cmd := fmt.Sprintf("~ %02d %02X ", s.channel, c)
+	cmd = cmd + checksum(cmd[1:]) + "\r"
+
+	_, err := fmt.Fprint(s.rw, cmd)
+	if err != nil {
+		return "", fmt.Errorf("failed writing command to read writer: %s", err)
+	}
+
+	scanner := bufio.NewScanner(s.rw)
+	scanner.Split(split)
+
+	if !scanner.Scan() {
+		return "", fmt.Errorf("failed reading to end of command response: %s", scanner.Err())
+	}
+
+	response := scanner.Text()
+
+	status := response[3:5]
+	code := response[6:8]
+	data := response[9 : len(response)-3]
+	csum := response[len(response)-2:]
+
+	if status != string(Ok) {
+		return "", fmt.Errorf("failed to execute command successful (response code: %s)", code)
+	}
+	if csum != checksum(response[:len(response)-2]) {
+		return "", fmt.Errorf("failed checksum of response")
+	}
+
+	return data, nil
+}
+
+func (s *SPCE) ReadFloat(c Code) (float64, error) {
+	data, err := s.Read(c)
+	if err != nil {
+		return 0, err
+	}
+
+	switch c {
```

Comment:
In principle, we can also remove the units for `Voltage` and `Pressure` and then have a single case handler to parse the float (`strconv.ParseFloat` also can handle ints)